### PR TITLE
Do not enable caching metrics by default and provide a guide

### LINKED ISSUE #16751: Do not enable caching metrics by default and provide a guide
### Description

We should not enable caching metrics by default but let users choose whether they want to expose metrics from caches.

We should also provide a guide about how to enable metrics.

### Discussion

https://github.com/keycloak/keycloak-documentation/pull/1753

### Motivation

Caching metrics should only be exposed if explicitly configured by users through the cache configuration file.

### Details

_No response_

### PR DESCRIPTION
**Pull request by pedroigor**
Closes #16751

As discussed at https://github.com/keycloak/keycloak-documentation/pull/1753 and with @ahus1:

* Do not automatically enable caching metrics but let users enable it explicitly if needed
* Added a metrics guide
* Update the caching guide

### CODE REVIEW

File: docs/guides/src/main/server/configuration-metrics.adoc [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+== Querying Metrics
+
+Keycloak exposes metrics at the following endpoint:
+
+* `/metrics`
+
+The response from the endpoint uses a `text/plain` content type and it is based on the Prometheus text format. The snippet bellow
+is an example of a response:
+
+[source, json]
```
Comment (ahus1) [inline docs/guides/src/main/server/configuration-metrics.adoc:28]: Nit-pick: it is actually plain text, and not JSON. As the current guides don't do syntax highlighting, this doesn't matter in the output. So I'll still approve it.

```suggestion
[source]
```

**Review by ahus1 — DISMISSED** [review]
I tested it and it now works as expected. Good to see that users can switch them on and off as needed. 

I also found that the metric names look the same as in KC20, and the way to enable metrics is also the same as in KC20 (that is: updating the Infinispan XML file, and adding it to the CLI). Will update the docs PR accordingly.

**Comment by sschu** [comment]
LGTM.

File: docs/guides/src/main/server/configuration-metrics.adoc (hunk 1 above)
Comment (pedroigor) [reply docs/guides/src/main/server/configuration-metrics.adoc]: Done

**Review by ahus1 — APPROVED** [review]

**Comment by pedroigor** [comment]
> I tested it and it now works as expected. Good to see that users can switch them on and off as needed.
> 
> I also found that the metric names look the same as in KC20, and the way to enable metrics is also the same as in KC20 (that is: updating the Infinispan XML file, and adding it to the CLI). Will update the docs PR accordingly.

I've a feeling that we would also need the same level of configuration for http metrics too. But we can do it later.

**Comment by ahus1** [comment]
@pedroigor - :+1: for configuring HTTP metrics in another issue/PR. BTW, the Quarkus setting for this would be `quarkus.micrometer.binder.http-server.enabled` AFAIK.

**Review by vmuzikar — APPROVED** [review]

**Pull request merged**